rust/libnewsboat/src/utils.rs

```diff
@@ -445,6 +445,24 @@ pub fn make_title(rs_str: String) -> String {
     }
 }
 
+/// Get the current working directory.
+/// Returns an empty string on error
+pub fn getcwd() -> String {
+    std::env::current_dir()
+        .and_then(|path| {
+            path.to_str()
+                .ok_or(std::io::Error::new(
+                    std::io::ErrorKind::Other,
+                    "Invalid unicode in path",
+                ))
+                .map(String::from)
+        })
+        .unwrap_or_else(|err| {
+            log!(Level::Warn, "Error getting current directory: {}", err);
+            "".to_owned()
```

Comment:
Should we really return an empty string on error like the C++ code does? IMO this isn't very rusty, on the other hand the function would basically end up calling just `std::env::current_dir` if we don't convert errors to empty strings.